packages/ember-glimmer/lib/syntax/curly-component.js

```diff
@@ -235,6 +235,10 @@ export class CurlyComponentDefinition extends ComponentDefinition {
     builder.attrs.static('class', 'ember-view');
   }
 
+  getComponentClass() {
```

Comment:
We don't have enough tests to cover mixing up stuff for the caching we are doing because these issues are not being caught